### PR TITLE
fix(backups): no full backup

### PR DESCRIPTION
**Pull request by AdityaNPL**
- [x] test on a test DB

### CODE REVIEW

File: src/AgDatabase.cs [hunk 1]
```diff
@@ -131,9 +131,19 @@ public List<BackupMetadata> RecentBackups()
...
+        try
+        {
+          fullBackupLsnBag.Add(s.Database(Name).MostRecentFullBackupLsn());
+        }
+        catch { }
+      });
 
       // find all backups in that chain
+      if (fullBackupLsnBag.Count == 0)
+        throw new Exception("Could not find any full backups");
```
Comment (lotsofspots) [inline src/AgDatabase.cs:145]: Do we want to append the Name here?

Comment (AdityaNPL) [reply src/AgDatabase.cs:145]: can do

File: src/AgDatabase.cs [hunk 2]
```diff
@@ -131,9 +131,19 @@ public List<BackupMetadata> RecentBackups()
...
+      {
+        try
+        {
+          fullBackupLsnBag.Add(s.Database(Name).MostRecentFullBackupLsn());
+        }
+        catch { }
+      });
 
       // find all backups in that chain
+      if (fullBackupLsnBag.Count == 0)
```
Comment (mja9) [inline src/AgDatabase.cs:144]: Is there a `fullBackupLsnBag.IsEmpty()`?

**Review by Kaneraz — APPROVED** [review]
Looks correct. 

When throwing a specific exception I prefer to make a new type, and catch it rather than just throw exception. 

Also, it can be expected that a backup may not exist on each node, so the interface should change to be nullable. Save the exceptions for when something unexpected happens.

File: src/AgDatabase.cs (hunk 2 above)
Comment (AdityaNPL) [reply src/AgDatabase.cs]: yup, looks like there is - changed it 👍🏽

**Comment by AdityaNPL** [comment]
@Kaneraz - I agree, maybe we can return `decimal?` or some other nullable value instead.
I need this fix right now so will merge it in and follow up with a refactor tomorrow.

**Pull request merged**